cmd/nri-kubernetes/main.go

```diff
@@ -3,152 +3,101 @@ package main
 import (
 	"context"
 	"fmt"
+	"os"
 	"time"
 
 	"github.com/newrelic/infra-integrations-sdk/integration"
 	"github.com/newrelic/infra-integrations-sdk/log"
 	"github.com/sethgrid/pester"
-	"k8s.io/apimachinery/pkg/version"
 
 	"github.com/newrelic/nri-kubernetes/v2/internal/config"
-	"github.com/newrelic/nri-kubernetes/v2/internal/discovery"
-	"github.com/newrelic/nri-kubernetes/v2/src/apiserver"
 	"github.com/newrelic/nri-kubernetes/v2/src/client"
 	"github.com/newrelic/nri-kubernetes/v2/src/ksm"
-	"github.com/newrelic/nri-kubernetes/v2/src/metric"
-	"github.com/newrelic/nri-kubernetes/v2/src/scrape"
+	ksmClient "github.com/newrelic/nri-kubernetes/v2/src/ksm/client"
 	"github.com/newrelic/nri-kubernetes/v2/src/sink"
-	"github.com/newrelic/nri-kubernetes/v2/src/storage"
 )
 
-const defaultLabelSelector = "app.kubernetes.io/name=kube-state-metrics"
+var logger log.Logger
 
-type clientsCluster struct {
-	k8s                 client.Kubernetes
-	ksm                 ksm.Client
-	api                 apiserver.Client
-	endpointsDiscoverer discovery.EndpointsDiscoverer
-	servicesDiscoverer  discovery.ServiceDiscoverer
+const (
+	ExitClients = iota
+	ExitIntegration
+	ExitLoop
+)
+
+type clusterClients struct {
+	k8s client.Kubernetes
+	ksm ksmClient.Client
 }
 
 func main() {
 	c := config.LoadConfig()
-	logger := log.NewStdErr(c.Verbose)
+	logger = log.NewStdErr(c.Verbose)
 
-	k8s, err := client.NewKubernetes(true)
+	// TODO: Can this error?
+	conf := config.LoadConfig()
```

Comment:
Since we are going to parse a file, yes, I imaging it will return an error as well

Reply:
Will be taken care of in the config PR.